src/views/AnnualReport.vue

```diff
@@ -506,33 +502,46 @@ export default {
       }
     }
 
-    // NB: filing id of 0 means "new AR"
-    // otherwise it's a draft AR filing id
-    this.filingId = +this.$route.params.filingId // number (may be NaN)
+    // Filing ID may be 0, N or NaN
+    this.filingId = +this.$route.params.filingId
+  },
 
+  mounted (): void {
     // if tombstone data isn't set, go back to dashboard
     if (!this.entityIncNo || !this.ARFilingYear || isNaN(this.filingId)) {
+      console.log('Annual Report error - missing Entity Inc No, AR Filing Year, or Filing ID!')
       this.$router.push({ name: Routes.DASHBOARD })
+      return // don't continue
+    }
+    if (this.isCoop && (!this.arMinDate || !this.arMaxDate)) {
+      console.log('Annual Report error - missing AR Min Date or AR Max Date!')
+      this.$router.push({ name: Routes.DASHBOARD })
+      return // don't continue
+    }
+    if (this.isBComp && !this.nextARDate) {
+      console.log('Annual Report error - missing Next AR Date!')
+      this.$router.push({ name: Routes.DASHBOARD })
+      return // don't continue
     }
-  },
 
-  mounted (): void {
     // wait until entire view is rendered (including all child components)
     // see https://v3.vuejs.org/api/options-lifecycle-hooks.html#mounted
     this.$nextTick(async () => {
       if (this.filingId > 0) {
-        this.loadingMessage = `Resuming Your ${this.ARFilingYear} Annual Report`
         // resume draft filing
+        this.loadingMessage = `Resuming Your ${this.ARFilingYear} Annual Report`
         await this.fetchDraftFiling()
         // fetch original office addresses and directors
         // update working data only if it wasn't in the draft
         if (!this.isJestRunning) {
-          await this.$refs.officeAddressesComponent.getOrigAddresses(this.asOfDate, isEmpty(this.updatedAddresses))
+          const isEmptyAddresses =
```

Comment:
This is a bug fix. `updatedAddresses` was never empty because it contains (empty) objects. The result was that addresses were not reloaded when a draft AR was resumed. It now works correctly.

This same fix was made in Standalone Office Address Filing page.